### DISCUSSION TITLE
is defining the GOOGLE_CLOUD_CPP_OPENTELEMETRY_TRACING environment variable all I need?

### DISCUSSION THREAD

**Discussion by willie** [googleapis/google-cloud-cpp#13765] [Q&A] [answered]
I have a C++ program that I recently added OpenTelemetry instrumentation to. Same program uses Cloud Storage, as well. In vcpkg (very recent version), I added the `opentelemetry` feature to `google-cloud-cpp` in my manifest, and I've defined `GOOGLE_CLOUD_CPP_OPENTELEMETRY_TRACING` as an environmental variable. 

I've not changed any other code, as I've already defined a global exporter, etc. in my existing code. I'm not sending my Otel data to Cloud Trace, FWIW, but I'm seeing all of my traces, etc. in Uptrace where I have things configured, so I'm pretty sure that isn't the problem.

Is there more I have to do to make this work?

**Comment by dbolduc** [comment, marked as answer]
Argh, sorry, this is a bug. Thanks for the report.

In the latest version of `google-cloud-cpp`, the `storage` and `bigtable` libraries do not respect the `GOOGLE_CLOUD_CPP_OPENTELEMETRY_TRACING` variable. This is not by design. It was an oversight.

I fixed this for `bigtable` in #13748. I will send a PR to fix `storage`.

I think your options are to:

A. Explicitly set `OpenTelemetryTracingOption` in your client. e.g.

https://github.com/googleapis/google-cloud-cpp/blob/8ca67ec675ae95121fd00c7bba0a9595cce721a9/google/cloud/opentelemetry/quickstart/quickstart.cc#L38-L40

B. Wait for us to make the fix, release a new version of `google-cloud-cpp`, and push that version to `vcpkg`. This will not get completed until April.

Sorry for the trouble.

**Reply by willie** [reply]
I appreciate the reply and workaround! Option A immediately solved my problem. Thank you!

**Reply by dbolduc** [reply]
Cool. I am glad your problem is solved!

FWIW, I sent #13766 to fix the handling in the `storage::Client`.
